### DISCUSSION TITLE
Sign In with Microsoft

### DISCUSSION THREAD

**Discussion by Chatano** [invertase/react-native-firebase#5401] [Q&A]
Hello! I have been trying to authenticate with microsoft using firebase but I got a few problems...
When my user authenticate with microsoft using the lib 'react-native-azure-auth' I receive some tokens (highlighting the rawIdToken) but then, when I try to create some credential or use the custom token at firebase I have no results.

My objective is to authenticate the user with microsoft and create a firebase token with it, I tried with the custom token and create credentials methods, but nothing has worked so far, which data should I use from 'react-native-azure-auth'(or other lib that you recommend) to authenticate with firebase? And which method should I use from firebase auth?

The documentation seems incomplete about Microsoft Authentication.

**Comment by joaonew** [comment]
I have the same problem

**Comment by mikehardy** [comment]
It might be easier - since the underlying firebase-ios-sdk / firebase-android-sdk both support it, to actually implement native support for it here as a PR

At first look that might not seem true, but let me be specific:

1) Native support: ios https://firebase.google.com/docs/auth/ios/microsoft-oauth#objective-c_3 / android https://firebase.google.com/docs/auth/android/microsoft-oauth

2) our testing app is what we use to develop features, it only takes a few steps to get setup: https://github.com/invertase/react-native-firebase/blob/master/tests/README.md

3) adding an auth provider is not that difficult, here's a PR to the last one added (apple auth) - it shows the general shape of a PR to add one https://github.com/invertase/react-native-firebase/pull/2979/files though it was an OAuth provider so this will be different. If you follow a more similar example though (like Google provider perhaps) through from index.js/index.d.ts through the provider javascript, down to the android / ios implementation it doesn't look so bad? https://github.com/invertase/react-native-firebase/tree/master/packages/auth

So that is the real answer to "or other lib that you recommend" - I would recommend actually adding it as a real provider here.

Other than that, I'm not sure, handling auth tokens etc between libs is always an exacting thing and I have no experience with Azure unfortunately.

**Reply by mikehardy** [reply]
Oh - worth mentioning that when you develop using the tests app, the native files are editable in their actual packages so you can use Xcode / Android Studio easily on the real files and the just rebuild / reinstall the app, it's pretty seemless. Then when it appears to be working you can post a PR and - this is important - all our PRs run a github action to generate a set of patch-package compatible patches so you can easily try it in your real project without waiting for anyone to build something and without trying to figure out how to integrate a monorepo with changes into your own app - example here https://github.com/invertase/react-native-firebase/actions/runs/857909452

**Reply by Chatano** [reply]
Oh, I have more experience with js than native code, do you have some estimated time to create a new method to simplificate this process or not, like creating a microsoft credential? I'm thinking about using a webview on the app and sign in at it in the mean time
